DeleteUserDataActivity/__tests__/authenticationLockService.test.ts

```diff
@@ -0,0 +1,157 @@
+
+import AuthenticationLockService from "../authenticationLockService";
+
+import {
+  aFiscalCode,
+  aNotReleasedData,
+  anUnlockCode,
+  anothernUnlockCode,
+  brokeEntityProfileLockedRecordIterator,
+  errorProfileLockedRecordIterator,
+  getProfileLockedRecordIterator,
+  listLockedProfileEntitiesMock,
+  lockedProfileTableClient,
+  submitTransactionMock
+} from "../../__mocks__/lockedProfileTableClient";
+import { UnlockCode } from "../../generated/definitions/UnlockCode";
+
+// -------------------------------------------
+// Variables
+// -------------------------------------------
+
+const service = new AuthenticationLockService(lockedProfileTableClient);
+
+describe("AuthenticationLockService#getAllUserAuthenticationLockData", () => {
+  it("should return an empty array if query returns no records from table storage", async () => {
+    const result = await service.getAllUserAuthenticationLockData(
+      aFiscalCode
+    )();
+
+    expect(result).toEqual(E.right([]));
+    expect(listLockedProfileEntitiesMock).toHaveBeenCalledWith({
+      queryOptions: {
+        filter: `PartitionKey eq '${aFiscalCode}'`
+      }
+    });
+  });
+
+  it.each`
+    title             | records
+    ${"one record"}   | ${[aNotReleasedData]}
+    ${"more records"} | ${[aNotReleasedData, { ...aNotReleasedData, rowKey: anothernUnlockCode, Released: true }]}
+  `(
+    "should return all the records, if $title are found in table storage",
+    async ({ records }) => {
+      listLockedProfileEntitiesMock.mockImplementationOnce(() =>
+        getProfileLockedRecordIterator(records)
+      );
+
+      const result = await service.getAllUserAuthenticationLockData(
+        aFiscalCode
+      )();
+
+      expect(result).toEqual(E.right(records));
+    }
+  );
+
+  it("should return an error if something went wrong retrieving the records", async () => {
+    listLockedProfileEntitiesMock.mockImplementationOnce(
+      errorProfileLockedRecordIterator
+    );
+
+    const result = await service.getAllUserAuthenticationLockData(
+      aFiscalCode
+    )();
+
+    expect(result).toEqual(E.left(Error("an Error")));
+  });
+
+  it("should return an error if something went wrong decoding a record", async () => {
+    listLockedProfileEntitiesMock.mockImplementationOnce(
+      brokeEntityProfileLockedRecordIterator
+    );
+
+    const result = await service.getAllUserAuthenticationLockData(
+      aFiscalCode
+    )();
+
+    expect(result).toEqual(
+      E.left(
+        Error(
+          'value ["CF"] at [root.0.0.0.partitionKey] is not a valid [string that matches the pattern "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"] / value [undefined] at [root.0.0.1.Released] is not a valid [boolean] / value ["CF"] at [root.0.1.partitionKey] is not a valid [string that matches the pattern "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"]'
+        )
+      )
+    );
+  });
```

Comment:
I think this can replace the long error pasted 
```suggestion
    expect(result).toMatchObject({
      _tag: "Left",
      left: {
        message: expect.stringContaining(
          "is not a valid [string that matches the pattern"
        )
      }
    });
```

Reply:
Done in [b5fe4ea](https://github.com/pagopa/io-functions-admin/pull/226/commits/b5fe4ea584361c53e37ba5b97ea4113e7136accd)